dbms/src/Dictionaries/RangeHashedDictionary.cpp

```diff
@@ -1,7 +1,46 @@
 #include <Dictionaries/RangeHashedDictionary.h>
 #include <Dictionaries/RangeDictionaryBlockInputStream.h>
+#include <Functions/FunctionHelpers.h>
+#include <Columns/ColumnNullable.h>
+#include <ext/range.h>
 
 
+namespace
+{
+
+using RangeStorageType = DB::RangeHashedDictionary::RangeStorageType;
+
+// Null values mean that specified boundary, either min or max is not set on range.
+// To simplify comparison, null value of min bound should be bigger than any other value,
+// and null value of maxbound - less than any value.
+const RangeStorageType RANGE_MIN_NULL_VALUE = std::numeric_limits<RangeStorageType>::max();
+const RangeStorageType RANGE_MAX_NULL_VALUE = std::numeric_limits<RangeStorageType>::min();
+
+// Handle both kinds of null values: explicit nulls of NullableColumn and 'implicit' nulls of Date type.
+RangeStorageType getColumnIntValueOrDefault(const DB::IColumn & column, size_t index, bool isDate, const RangeStorageType & default_value)
+{
+    if (column.isNullAt(index))
+        return default_value;
+
+    const RangeStorageType result = static_cast<RangeStorageType>(column.getInt(index));
+    if (isDate && !DB::RangeHashedDictionary::Range::isCorrectDate(result))
+        return default_value;
+
+    return result;
+}
+
+const DB::IColumn & unwrapNullableColumn(const DB::IColumn & column)
```

Comment:
Ok. BTW, there are
`makeNullable` and `removeNullable` functions in `DataTypeNullable.h`
and `makeNullable` in `ColumnNullable.h`.

We can make it complete by adding `removeNullable` in `ColumnNullable.h`.